src/replication/adapter/native/adapter.go

```diff
@@ -159,37 +162,75 @@ func (a *Adapter) FetchImages(filters []*model.Filter) ([]*model.Resource, error
 		}
 	}
 
-	var resources []*model.Resource
-	for _, repository := range repositories {
-		vTags, err := a.getVTags(repository.Name)
-		if err != nil {
-			return nil, err
-		}
-		if len(vTags) == 0 {
-			continue
-		}
-		for _, filter := range filters {
-			if err = filter.DoFilter(&vTags); err != nil {
-				return nil, err
+	rawResources := make([]*model.Resource, len(repositories))
+	var wg = new(sync.WaitGroup)
+	ctx, cancel := context.WithCancel(context.Background())
+	var passportsPool = utils.NewPassportsPool(adp.MaxConcurrency, ctx.Done())
+
+	for i, r := range repositories {
```

Comment:
Could we write a function for this for_loop? it seems we have 3 sections are similar code.

Reply:
+1,

Can we move those duplicated codes into a unified runner? Seems most of the code are similar.

Reply:
OK